test/es-module/test-esm-type-flag-package-scopes.mjs

```diff
@@ -0,0 +1,150 @@
+             strictEqual(code, 0);
+             strictEqual(signal, null);
+           });
+
+           it('should import an extensionless JavaScript file within a "type": "module" scope', async () => {
+             const { default: defaultExport } =
+              await import(fixtures.fileURL('es-modules/package-type-module/noext-esm'));
+             strictEqual(defaultExport, 'module');
+           });
+
+           it('should run as Wasm an extensionless Wasm file within a "type": "module" scope', async () => {
+             const { code, signal, stdout, stderr } = await spawnPromisified(process.execPath, [
+               '--experimental-type=module',
+               '--experimental-wasm-modules',
+               '--no-warnings',
+               fixtures.path('es-modules/package-type-module/noext-wasm'),
+             ]);
+
+             strictEqual(stderr, '');
+             strictEqual(stdout, 'executed\n');
+             strictEqual(code, 0);
+             strictEqual(signal, null);
+           });
+
+           it('should import as Wasm an extensionless Wasm file within a "type": "module" scope', async () => {
+             const { add } = await import(fixtures.fileURL('es-modules/package-type-module/noext-wasm'));
+             strictEqual(add(1, 2), 3);
+           });
+         });
+
+describe(`the type flag should change the interpretation of certain files within a package scope that lacks a
+"type" field and is not under node_modules`, { concurrency: true }, () => {
+  it('should run as ESM a .js file within package scope that has no defined "type" and is not under node_modules',
+     async () => {
+       const { code, signal, stdout, stderr } = await spawnPromisified(process.execPath, [
+         '--experimental-type=module',
+         fixtures.path('es-modules/package-without-type/module.js'),
+       ]);
+
+       strictEqual(stderr, '');
+       strictEqual(stdout, 'executed\n');
+       strictEqual(code, 0);
+       strictEqual(signal, null);
+     });
+
+  it(`should run as ESM an extensionless JavaScript file within a package scope that has no defined "type" and is not
+under node_modules`, async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(process.execPath, [
+      '--experimental-type=module',
+      fixtures.path('es-modules/package-without-type/noext-esm'),
+    ]);
+
+    strictEqual(stderr, '');
+    strictEqual(stdout, 'executed\n');
+    strictEqual(code, 0);
+    strictEqual(signal, null);
+  });
+
+  it(`should run as Wasm an extensionless Wasm file within a package scope that has no defined "type" and is not under
+  node_modules`, async () => {
+    const { code, signal, stdout, stderr } = await spawnPromisified(process.execPath, [
+      '--experimental-type=module',
+      '--experimental-wasm-modules',
+      '--no-warnings',
+      fixtures.path('es-modules/noext-wasm'),
+    ]);
+
+    strictEqual(stderr, '');
+    strictEqual(stdout, '');
```

Comment:
So while this validates that there are no error output and exit code is 0, it would be nice to have some kind of confirmation that the code have actually run (hence my suggestion of adding `console.log` to the JS dependency)

Reply:
I added it in [`62b4b09` (#49869)](https://github.com/nodejs/node/pull/49869/commits/62b4b0912094f39ed0baf456d9f933780447ad66), I can look into adding it for this one too.

Reply:
This one was excluded because this is just running the Wasm file directly as the entry point. Is there a way to print to stdout from within Wasm?

Reply:
I’ve been looking into that, and it doesn’t seem to be possible without involving JS. Maybe there’s a way to import `node:console` somehow, but if we’re not able to figure out, it’s not blocking.

Reply:
Happy to add it as a future improvement in a follow-up.